OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -210,5 +214,13 @@ public static void InflictDamage(this Actor self, Actor attacker, int damage, IW
 			if (health == null) return;
 			health.InflictDamage(self, attacker, damage, warhead, false);
 		}
+
+		public static void Kill(this Actor self, Actor attacker)
+		{
+			if (self.Disposed) return;
+			var health = self.TraitOrDefault<Health>();
```

Comment:
You can save a trait lookup by keeping Kill on the actor and then using the cached health reference there.